packages/diagrams/frontend/sirius-components-diagrams/src/renderer/edge/EdgeLayout.ts

```diff
@@ -86,8 +88,40 @@ const isVerticalLayoutDirection = (layoutDirection: string): boolean =>
   layoutDirection === 'DOWN' || layoutDirection === 'UP';
 const isHorizontalLayoutDirection = (layoutDirection: string): boolean =>
   layoutDirection === 'RIGHT' || layoutDirection === 'LEFT';
+const isPositionDependentRotation = (nodeData: NodeData): boolean => {
+  if ('positionDependentRotation' in nodeData) {
+    return (nodeData as FreeFormNodeData).positionDependentRotation;
+  }
+  return false;
+};
+const computeBorderNodeHandlePosition = (borderNodeData: NodeData, isInside: boolean): Position => {
+  let borderNodePosition: BorderNodePosition | null = borderNodeData.borderNodePosition;
+  if (isPositionDependentRotation(borderNodeData)) {
+    borderNodePosition = BorderNodePosition.WEST;
+  }
+  switch (borderNodePosition) {
+    case BorderNodePosition.EAST:
+      return isInside ? Position.Left : Position.Right;
+    case BorderNodePosition.SOUTH:
+      return isInside ? Position.Top : Position.Bottom;
+    case BorderNodePosition.WEST:
+      return isInside ? Position.Right : Position.Left;
+    case BorderNodePosition.NORTH:
+      return isInside ? Position.Bottom : Position.Top;
+    default:
+      return isInside ? Position.Left : Position.Right;
+  }
+};
 
 const getParameters: GetParameters = (movingNode, nodeA, nodeB, visiblesNodes, layoutDirection) => {
```

Comment:
Could you rename _nodeA_ and _nodeB_ by something more meaningful please?

Reply:
I don't want to introduce a hierarchy between nodeA and nodeB (for example, naming them sourceNode and targetNode) because we need to calculate the position for both edge nodes. So in one case, the edge's targetNode would become the sourceNode of this function. I think that's why we named the two nodes A and B.